src/components/dashboard/SubscriptionExpirationModal.jsx

```diff
@@ -105,22 +134,24 @@ const SubscriptionExpirationModal = () => {
   );
 
   // If the subscription has already expired, we show a different un-dismissible modal
-  const subscriptionExpired = daysUntilExpiration <= SUBSCRIPTION_EXPIRED;
-  if (subscriptionExpired) {
+  if (!isCurrent && !hasLocalStorageData(EXPIRED_SUBSCRIPTION_MODAL_LOCALSTORAGE_KEY(subscriptionLicense))) {
     return (
       <Modal
         dialogClassName={`${MODAL_DIALOG_CLASS_NAME} expired`}
         renderHeaderCloseButton={false}
         title={renderTitle()}
         body={renderExpiredBody()}
         closeText="OK"
-        onClose={() => {}}
+        onClose={() => {
+          setLocalStorageData([EXPIRED_SUBSCRIPTION_MODAL_LOCALSTORAGE_KEY(subscriptionLicense), true]);
+        }}
         open
+        data-testid="expiration-modal"
       />
     );
   }
 
-  if (daysUntilExpiration > SUBSCRIPTION_DAYS_REMAINING_SEVERE) {
+  if (daysUntilExpirationIncludingRenewals > SUBSCRIPTION_DAYS_REMAINING_SEVERE || !isCurrent) {
```

Comment:
So the reason for the additional gating is to prevent cases where already expired subscriptions falls into the same category of being displayed by the expiration modal. Removing the logic of `!isCurrent` and having a subscription that is expired displays the following modal after you refresh once you've dismissed the expired modal. 
![Screenshot 2024-05-13 at 10 56 44 AM](https://github.com/openedx/frontend-app-learner-portal-enterprise/assets/82611798/384c06fe-3b09-4a49-ac4e-179c2578d791)

Reply:
Thanks for clarifying; I understand the behavior now after replicating locally. I wonder if it would improve code readability if the `!isCurrent && !seenExpiredSubscriptionModal` conditional was broken into a nested conditional?

```jsx
if (!isCurrent) {
  // Expired modal already seen; render nothing.
  if (seenExpiredSubscriptionModal) {
    return null;
  }
  return (
    <Modal ... />
  );
}
```

That way, `if (daysUntilExpirationIncludingRenewals > SUBSCRIPTION_DAYS_REMAINING_SEVERE) {` below doesn't need to take into account `!isCurrent` given its handled in the above conditionals. This proposed change might be a slight improvement in that this conditional no longer deals with both expired and non-expired conditions. No strong opinions here, though.